httpx/client.py

```diff
@@ -601,6 +605,7 @@ def redirect_stream(
             else:
                 request = next_request
                 await response.close()
+            requested_more_than_once = True
```

Comment:
Nit: should we keep this in the `else` block, since it represents what happens when we're asked to make a new request?

```suggestion
                requested_more_than_once = True
```